runtimes/pythonrt/py-sdk/autokitteh/atlassian.py

```diff
@@ -87,6 +87,82 @@ def __atlassian_jira_client_cloud_oauth2(connection: str, **kwargs):
     )
 
 
+def confluence_client(connection: str, **kwargs):
+    """Initialize an Atlassian Confluence client, based on an AutoKitteh connection.
+
+    API reference:
+    https://atlassian-python-api.readthedocs.io/confluence.html
+
+    Code samples:
+    https://github.com/atlassian-api/atlassian-python-api/tree/master/examples/confluence
+
+    Args:
+        connection: AutoKitteh connection name.
+
+    Returns:
+        Atlassian-Python-API Confluence client.
+
+    Raises:
+        ValueError: AutoKitteh connection name is invalid.
+        RuntimeError: OAuth 2.0 access token expired.
+        ConnectionInitError: AutoKitteh connection was not initialized yet.
+        EnvVarError: Required environment variable is missing or invalid.
+    """
+    check_connection_name(connection)
+
+    if os.getenv(connection + "__oauth_AccessToken"):
+        return __atlassian_confluence_client_cloud_oauth2(connection, **kwargs)
+
+    base_url = os.getenv(connection + "__BaseURL")
+    token = os.getenv(connection + "__Token")
+    if token:
+        email = os.getenv(connection + "__Email")
+        if not email:
+            return Confluence(url=base_url, token=token, **kwargs)
+        return Confluence(
+            url=base_url,
+            username=email,
+            password=token,
+            cloud=True,
+            **kwargs,
+        )
+
+    raise ConnectionInitError(connection)
+
+
+def __atlassian_confluence_client_cloud_oauth2(connection: str, **kwargs):
+    """Initialize a Confluence client for Atlassian Cloud using OAuth 2.0."""
+    expiry = os.getenv(connection + "__oauth_Expiry")
+    if not expiry:
+        raise ConnectionInitError(connection)
+
+    # Convert Go's time string (e.g. "2024-06-20 19:18:17 -0700 PDT") to
```

Comment:
Why not use `datetime.strptime`?

Reply:
I copy and pasted existing code. There wasn't much additional thought behind it other than it's what we're already using.